src/libstd/collections/hash/map.rs

```diff
@@ -2445,13 +2444,25 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
-            Cell::new(sys::hashmap_random_keys())
+        thread_local!(static KEYS: Cell<[u64; 2]> = {
+            let mut buf = [0u8; 16];
+            // Use a constant seed on wasm32-unknown-unknown.
+            // `cfg(target = "..")` does not work right now, see:
+            // https://github.com/rust-lang/rust/issues/63217
+            if cfg!(not(all(
+                target_arch = "wasm32",
+                target_vendor = "unknown",
+                target_os = "unknown",
+            ))) {
+                getrandom::getrandom(&mut buf).unwrap();
+            }
```

Comment:
On `wasm32-unknown-unknown`, this will effectively be:

```rust
if false {
    getrandom::getrandom(&mut buf).unwrap();
}
```

... so it does require the `getrandom` dependency, even though this is dead code.

If you `#cfg(...)]` gate this expression instead, it won't always need `getrandom`.

Reply:
But IIUC Cargo determines crates to be compiled only by looking at Cargo.toml, without taking crate code into account. So even if a dependency is not used, it will be still compiled, which in our case means a compilation error.

I forgot that we can use `cfg` like this:
```rust
#[cfg(..)]
getrandom::getrandom(&mut buf).unwrap();
```
I will replace those lines in the next commit.